core/src/main/java/org/apache/accumulo/core/client/impl/Namespaces.java

```diff
@@ -100,50 +106,150 @@ private static ZooCache getZooCache(Instance instance) {
     return namespaceMap;
   }
 
-  public static boolean exists(Instance instance, String namespaceId) {
+  public static boolean exists(Instance instance, Namespace.ID namespaceId) {
     ZooCache zc = getZooCache(instance);
     List<String> namespaceIds = zc.getChildren(ZooUtil.getRoot(instance) + Constants.ZNAMESPACES);
-    return namespaceIds.contains(namespaceId);
-  }
-
-  public static String getNamespaceId(Instance instance, String namespace) throws NamespaceNotFoundException {
-    String id = getNameToIdMap(instance).get(namespace);
-    if (id == null)
-      throw new NamespaceNotFoundException(null, namespace, "getNamespaceId() failed to find namespace");
-    return id;
+    return namespaceIds.contains(namespaceId.canonicalID());
   }
 
+  /**
+   * @deprecated Do not use - String for namespace ID is not type safe. Use {@link #getNamespaceName(Instance, Namespace.ID)}
+   */
+  @Deprecated
```

Comment:
Rather than keep the old ones around, I would just change the implementation of the API methods to one which does something like:
```java
return Tables.getNameToIdMap().entrySet().stream()
        .collect(Collectors.toMap(Map.Entry::getKey,
                                  e -> e.canonicalId(),
                                  (v1,v2) ->{ throw new RuntimeException(String.format("Duplicate key for values %s and %s", v1, v2));},
                                  TreeMap::new)));
```